### PR TITLE
Add support for Apple specific extension (WebAuthn/App attestation)

### PR DESCRIPTION
**Pull request by agrinman**
This PR adds support for a particular Apple attestation X509 extension: https://developer.apple.com/documentation/devicecheck/validating_apps_that_connect_to_your_server
(OID 1.2.840.113635.100.8.2)

### CODE REVIEW

File: src/extensions.rs [hunk 1]
```diff
@@ -664,6 +669,26 @@ pub(crate) mod parser {
...
+    fn parse_sequence_single_octet_string<'a>(i: &'a [u8]) -> IResult<&'a [u8], &'a [u8], BerError> {
+        let (rest, obj) = parse_der_with_tag(i, BerTag::Sequence)?;
+        let seq = obj
+            .content
+            .as_sequence()
+            .or(Err(Err::Error(BerError::BerTypeError)))?
+            .first()
+            .ok_or(Err::Error(BerError::BerTypeError))?
+            .as_slice()
+            .or(Err(Err::Error(BerError::BerTypeError)))?;
```
Comment (chifflier) [inline src/extensions.rs:682]: This sequence is a bit complex and fragile (all errors will be the same). There is a function in `der-parser` that does exactly that, and could help rewriting this.
Is this extension a SEQUENCE containing an OCTET STRING? If yes, this code can be largely simplified.
The entire function could become something like:
```rust
    fn parse_sequence_single_octet_string<'a>(
        i: &'a [u8],
    ) -> IResult<&'a [u8], &'a [u8], BerError> {
        parse_ber_sequence_defined_g(|_, input| {
            map_res(parse_ber_octetstring, |x| x.as_slice())(input)
        })(i)
    }
```

I have no certificate with this extension. Could you test the code and update/resubmit your PR?

Comment (chifflier) [reply src/extensions.rs:682]: Note: the above code requires to add
```rust
use nom::combinator::map_res;
```

Comment (agrinman) [reply src/extensions.rs:682]: Thanks, that's awesome. Will add this shortly!

Comment (agrinman) [reply src/extensions.rs]: The above code didn't work for me and I am not sure why. I believe it's a sequence containing a single der encoded octet string. I did update the middle error above to indicate the sequence is missing an object...

Comment (chifflier) [reply src/extensions.rs]: I can't find a clear definition of this extension ... Best I could find is https://developer.apple.com/forums/thread/663118 which lets me think the sequence contains a tagged `[1]` value/

Can you provide some test data? Either the certificate, or an extract of the extension (for ex obtained using `openssl asn1parse`)

**Comment by cpu** [comment]
It looks like this branch hasn't had a push for 4 years and has collected some conflicts with `master` in the meantime. I'm going to close it since it will take some investment by an interested party to get it back into shape for review. 

If someone does come along with an interest please also try to find some test data so that test coverage could be added alongside the feature.

Thanks!

**Pull request closed without merging**